File: iconrotation/src/main/java/com/fhm/iconrotation/IconRotator.java
package com.fhm.iconrotation;

import android.view.animation.Animation;
import android.view.animation.DecelerateInterpolator;
import android.view.animation.RotateAnimation;
import android.widget.ImageView;

public class IconRotator {

    //initialRotationAngle = 0.0f, endRotationAngle = 90.0f, duration = 500
    public static void rotateToUpOrDown(ImageView ivRotate, float initialRotationAngle, float endRotationAngle, long duration){
        RotateAnimation rotateAnimation = new RotateAnimation(initialRotationAngle, endRotationAngle, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        rotateAnimation.setInterpolator(new DecelerateInterpolator());
        rotateAnimation.setDuration(duration);
        rotateAnimation.setFillAfter(true);
        ivRotate.startAnimation(rotateAnimation);
    }

}

File: iconrotation/src/main/java/com/fhm/iconrotation/IconRotator.kt
package com.fhm.iconrotation

import android.view.animation.Animation
import android.view.animation.DecelerateInterpolator
import android.view.animation.RotateAnimation
import android.widget.ImageView

class IconRotator {

    //initialRotationAngle = 0.0f, endRotationAngle = 90.0f, duration = 500
    fun rotateToUpOrDown(ivRotate: ImageView, initialRotationAngle: Float, endRotationAngle: Float, duration: Long) {
        val rotateAnimation = RotateAnimation(initialRotationAngle, endRotationAngle, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        rotateAnimation.interpolator = DecelerateInterpolator()
        rotateAnimation.duration = duration
        rotateAnimation.fillAfter = true
        ivRotate.startAnimation(rotateAnimation)
    }

}